test/metabase/api/public_test.clj

```diff
@@ -460,6 +460,38 @@
                        :parameters}
                      (set (keys public-action))))))))))))
 
+(deftest public-dashboard-param-link-to-a-field-without-full-field-values-test
```

Comment:
We don't need this test, given it is almost the same as the one in `test/metabase/api/dashboard_test.clj`, and it's not testing anything special about public dashboards vs normal dashboards. We can assume that public dashboards reuse the code for normal dashboards when it comes to fetching field values. I would exclude it purely for the cost of having to maintain it in the future.

Reply:
Let me share my reasoning on why I've added that test. 

GET request on public dashboard is processed with `metabase.api.public/public-dashboard`. This function sets `metabase.models.params/*ignore-current-user-perms-and-return-all-field-values*` to true [here](https://github.com/metabase/metabase/blob/e55e7f78c74526472e8b5aa4996a45e9157cb445/src/metabase/api/public.clj#L234). As a consequence `metabase.models.params/field-ids->param-field-values` calls `metabase.models.params/field-ids->param-field-values-ignoring-current-user` during values hydration [here](https://github.com/metabase/metabase/blob/e55e7f78c74526472e8b5aa4996a45e9157cb445/src/metabase/models/params.clj#L91). Alternatively, hydration of non-public dashboard's values goes through `params.field-values/field-id->field-values-for-current-user`.

So that test ensures that `field-ids->param-field-values-ignoring-current-user` works correctly in context of request processing.

On the other hand, another test to maintain just because of that may be overkill. Please just confirm that test should be removed, if that is the case, and I will remove it shortly.

Reply:
You're correct, my bad. I didn't see that!